sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -172,21 +172,12 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
 
       CSVRelation.dropHeaderLine(file, lineIterator, csvOptions)
 
-      val csvParser = new CsvReader(csvOptions)
-      val tokenizedIterator = lineIterator.filter { line =>
+      val linesWithoutHeader = lineIterator.filter { line =>
```

Comment:
It is dropped here `CSVRelation.dropHeaderLine(file, lineIterator, csvOptions)` but it just consumes the header. Let me just leave it as a comment.